packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx

```diff
@@ -0,0 +1,192 @@
+  )
+  const [solanaListenSetting, setSolanaListenSetting] = useState(
+    getOverrideSetting(FeatureFlags.SOLANA_LISTEN_ENABLED)
+  )
+  const [solWalletAudioSetting, setSolWalletAudioSetting] = useState(
+    getOverrideSetting(FeatureFlags.SOL_WALLET_AUDIO_ENABLED)
+  )
+  const [surfaceAudioSetting, setSurfaceAudioSetting] = useState(
+    getOverrideSetting(FeatureFlags.SURFACE_AUDIO_ENABLED)
+  )
+  const [tippingSetting, setTippingSetting] = useState(
+    getOverrideSetting(FeatureFlags.TIPPING_ENABLED)
+  )
+
+  type FlagInfo = {
+    label: string
+    val: OverrideSetting
+    setter: (val: OverrideSetting) => void
+  }
+
+  const flagInfoMap: Record<FeatureFlags, FlagInfo> = {
+    [FeatureFlags.CHALLENGE_REWARDS_UI]: {
+      label: 'Challenge Rewards UI',
+      val: challengeRewardsUiSetting,
+      setter: setChallengeRewardsUiSetting
+    },
+    [FeatureFlags.DISABLE_SIGN_UP_CONFIRMATION]: {
+      label: 'Disable Sign Up Confirmation',
+      val: signUpConfirmationSetting,
+      setter: setSignUpConfirmationSetting
+    },
+    [FeatureFlags.ENABLE_SPL_AUDIO]: {
+      label: 'Enable SPL Audio',
+      val: splAudioSetting,
+      setter: setSplAudioSetting
+    },
+    [FeatureFlags.PLAYLIST_FOLDERS]: {
+      label: 'Playlist Folders',
+      val: playlistFoldersSetting,
+      setter: setPlaylistFoldersSetting
+    },
+    [FeatureFlags.PLAYLIST_UPDATES_ENABLED]: {
+      label: 'Playlist Updates Enabled',
+      val: playlistUpdatesSetting,
+      setter: setPlaylistUpdatesSetting
+    },
+    [FeatureFlags.PREFER_HIGHER_PATCH_FOR_PRIMARY]: {
+      label: 'Prefer Higher Patch for Primary',
+      val: primaryPatchSetting,
+      setter: setPrimaryPatchSetting
+    },
+    [FeatureFlags.PREFER_HIGHER_PATCH_FOR_SECONDARIES]: {
+      label: 'Prefer Higher Patch for Secondaries',
+      val: secondaryPatchSetting,
+      setter: setSecondaryPatchSetting
+    },
+    [FeatureFlags.SHARE_SOUND_TO_TIKTOK]: {
+      label: 'Share Sound to TikTok',
+      val: tikTokSoundSetting,
+      setter: setTikTokSoundSetting
+    },
+    [FeatureFlags.SOLANA_LISTEN_ENABLED]: {
+      label: 'Solana Listen Enabled',
+      val: solanaListenSetting,
+      setter: setSolanaListenSetting
+    },
+    [FeatureFlags.SOL_WALLET_AUDIO_ENABLED]: {
+      label: 'Sol Wallet Audio Enabled',
+      val: solWalletAudioSetting,
+      setter: setSolWalletAudioSetting
+    },
+    [FeatureFlags.SURFACE_AUDIO_ENABLED]: {
+      label: 'Surface Audio Enabled',
+      val: surfaceAudioSetting,
+      setter: setSurfaceAudioSetting
+    },
+    [FeatureFlags.TIPPING_ENABLED]: {
+      label: 'Tipping Enabled',
+      val: tippingSetting as OverrideSetting,
+      setter: setTippingSetting
+    }
+  }
+
+  const openModal = () => setIsOpen(true)
+  const closeModal = () => setIsOpen(false)
+
+  useEffect(() => {
+    window.addEventListener('keypress', (e: KeyboardEvent) => {
+      if (e.ctrlKey && e.key === 'f') isOpen ? closeModal() : openModal()
```

Comment:
Kinda worried about this being too accessible for the general public. I think we should do it like the discovery node selection popup does: https://github.com/AudiusProject/audius-client/blob/main/packages/web/src/components/discovery-node-selection/DiscoveryNodeSelection.tsx#L13

That requires the dev mode local storage item to be set for the hotkey to work

Reply:
+1